tests/test_flow/test_reloading.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+import tempfile
+import sys
+import time
+
+
+@pytest.fixture(scope="function")
+def make_tmp_flow_module():
+    # We want to be able to do import statement on the temporary module, so we
+    # create a temporary directory here and add it to sys.path.
+    tmp_dir = tempfile.TemporaryDirectory()
+    sys.path.insert(0, tmp_dir.name)
+
+    def _make_tmp_flow_module(var_name, var_value):
+        with open(f"{tmp_dir.name}/tmp_flow_module.py", "w") as tmp_file:
+            tmp_file.write(
+                f"""
+import bionic as bn
+builder = bn.FlowBuilder('flow')
+builder.assign('x', {var_value})
+
+@builder
+def {var_name}(x):
+    '''documentation for {var_name}'''
+    return 2*x
+
+flow = builder.build()
+            """
+            )
+            tmp_file.flush()
+        # Needed to ensure that the file is loaded properly
+        # TODO: any way to avoid this?
+        time.sleep(1)
```

Comment:
As we discussed in the Hangout, it would be great to find out how to remove this `sleep`. If Naman or I can help with this, let us know.

Reply:
Turns out that pytest rewrites certain modules to provide more details during failed assertions. https://docs.pytest.org/en/stable/assert.html. The rewritten modules are stored in a cache.

https://github.com/pytest-dev/pytest/blob/d69abff2c7de8bc65b7f1ef867dec5b5b9c564bd/src/_pytest/assertion/rewrite.py#L379
Pytest does a timestamp check on the module files with second precision to determine whether the cache is valid. This is why the sleep with 1 second works.

The caching mechanism can be disabled.